### PR TITLE
Switch to new VMR control set

### PR DESCRIPTION
**Pull request by mmitche**
## Description

Now that fsharp is on 9.0, we can switch to the new control set. Generally:

- DotNetBuildFromSource -> DotNetBuildSourceOnly - Building a source-only build.
- DotnetBuildFromSourceFlavor == Product -> DotNetBuildOrchestrator == true - Building in the VMR, could be source-only or MS's build.
- ArcadeBuildFromSource -> DotNetBuildRepo == true -> Indicates an outer repo build.

Fixes # (issue, if applicable)

## Checklist

- [ ] Test cases added
- [ ] Performance benchmarks added in case of performance changes
- [ ] Release notes entry updated:
    > Please make sure to add an entry with short succinct description of the change as well as link to this pull request to the respective release notes file, if applicable.
    >
    > Release notes files:
    > - If anything under `src/Compiler` has been changed, please make sure to make an entry in `docs/release-notes/.FSharp.Compiler.Service/<version>.md`, where `<version>` is usually "highest" one, e.g. `42.8.200`
    > - If language feature was added (i.e. `LanguageFeatures.fsi` was changed), please add it to `docs/release-notes/.Language/preview.md`
    > - If a change to `FSharp.Core` was made, please make sure to edit `docs/release-notes/.FSharp.Core/<version>.md` where version is "highest" one, e.g. `8.0.200`.

    > Information about the release notes entries format can be found in the [documentation](https://fsharp.github.io/fsharp-compiler-docs/release-notes/About.html).
    > Example:
    > * More inlines for Result module. ([PR #16106](https://github.com/dotnet/fsharp/pull/16106))
    > * Correctly handle assembly imports with public key token of 0 length. ([Issue #16359](https://github.com/dotnet/fsharp/issues/16359), [PR #16363](https://github.com/dotnet/fsharp/pull/16363))
    > *`while!` ([Language suggestion #1038](https://github.com/fsharp/fslang-suggestions/issues/1038), [PR #14238](https://github.com/dotnet/fsharp/pull/14238))

    > **If you believe that release notes are not necessary for this PR, please add `NO_RELEASE_NOTES` label to the pull request.**

### CODE REVIEW

**Review by T-Gro — APPROVED** [review]

**Comment by psfinaki** [comment]
@mmitche so it's MIBC again 👀 

I introduced PGO to this repo a few weeks ago [here](https://github.com/dotnet/fsharp/pull/17513), it doesn't play well with source build (see the discussion in that PR), and we shouldn't ignore the packages either, so we disabled PGO-related logic for MIBC.

Given that in this PR the optimization packages pop up again, something doesn't play well together, IgnoreMIBC is not being set. I might look at this locally later today.

**Comment by psfinaki** [comment]
That's really weird. It looks like as if MSBuild wasn't setting this property up internally after receiving the new flags.

**Comment by mmitche** [comment]
I'll take a look too.

**Comment by psfinaki** [comment]
Alright, that seems to work - thanks @mmitche!

**Review by psfinaki — APPROVED** [review]

**Comment by psfinaki** [comment]
@dotnet/source-build-internal

File: eng/SourceBuild.props
Comment (MichaelSimons) [inline eng/SourceBuild.props:1]: Should this file also be renamed to the DotNetBuild.props as part of this work?

**Review by MichaelSimons — APPROVED** [review]

File: eng/SourceBuild.props
Comment (mmitche) [reply eng/SourceBuild.props:1]: Good call, thanks

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**